### DISCUSSION TITLE
how to use vuetify 3 alpha in new nuxt 3

### DISCUSSION THREAD

**Discussion by unknownman** [nuxt/nuxt#15929] [Questions]
Hi
I started using the new nuxt 3, but nowhere in the document or at the github I couldn't find a way to add vuetify. I add vuetify to buildModule in nuxt.config.js but it didnt work
I recevied this error: 

```
[Vue warn]: Failed to resolve component: v-row
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.

```

then when i add it with a plugin like below , it didnt work yet.
```
import { defineNuxtPlugin } from "#app";

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.app.config.compilerOptions.isCustomElement = (tag) =>
    tag.startsWith("v-"); // Doing something with nuxtApp
});

```
at all, how can we add vue 3 component in nuxt 3 , if you can mention a part of code i can prepare documentation too

thanks

**Comment by wobsoriano** [comment]


**Reply by unknownman** [reply]
it loads but again i have this warn

```
[Vue warn]: Failed to resolve component: v-row
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
(node:60851) UnhandledPromiseRejectionWarning: TypeError: Invalid value used as weak map key
    at WeakMap.set (<anonymous>)
    at normalizePropsOptions (/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2673:11)
    at createComponentInstance (/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6099:23)
    at renderComponentVNode (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:197:22)
    at Module.ssrRenderComponent (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:623:12)
    at _sfc_ssrRender (/.nuxt/dist/server/server.mjs:1685:31)
    at renderComponentSubTree (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:263:13)
    at renderComponentVNode (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:214:16)
    at Module.ssrRenderComponent (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:623:12)
    at default (file:///.nuxt/dist/server/server.mjs:1630:35)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:60851) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:60851) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Vue warn]: Failed to resolve component: v-row
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
(node:60851) UnhandledPromiseRejectionWarning: TypeError: Invalid value used as weak map key
    at WeakMap.set (<anonymous>)
    at normalizePropsOptions (/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:2673:11)
    at createComponentInstance (/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6099:23)
    at renderComponentVNode (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:197:22)
    at Module.ssrRenderComponent (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:623:12)
    at _sfc_ssrRender (file:///.nuxt/dist/server/server.mjs:1685:31)
    at renderComponentSubTree (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:263:13)
    at renderComponentVNode (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:214:16)
    at Module.ssrRenderComponent (/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:623:12)
    at default (file:///.nuxt/dist/server/server.mjs:1630:35)
(node:60851) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)

```

**Reply by wobsoriano** [reply]
Related to this https://github.com/nuxt/framework/discussions/1214

**Comment by wobsoriano** [comment]
Installation steps for Vuetify 3

Install vuetify 3

`yarn add vuetify@3.0.0-alpha.12`

Install sass

`yarn add sass -D`

Import components

```ts
// plugins/vuetify.ts
import { defineNuxtPlugin } from '#app'
import { createVuetify } from 'vuetify'
import {
    VApp,
    VAppBar,
    VBtn
} from 'vuetify/components'

// Import everything
// import * as components from 'vuetify/components'

export default defineNuxtPlugin((nuxtApp) => {
   const vuetify = createVuetify({
       components: {
           VApp,
           VAppBar,
           VBtn
       }
   })
   nuxtApp.vueApp.use(vuetify)
})
```

Transpile and add styles to `nuxt.config.ts`

```ts
export default defineNuxtConfig({
    css: ['vuetify/lib/styles/main.sass'],
    build: {
        transpile: ['vuetify']
    },
    vite: {
        define: {
            'process.env.DEBUG': 'false',
        }
    }
})
```

**Reply by Magnie** [reply]
I ran into an issue where the `sass` module wasn't available and had to run this to fix it:
```
# https://nuxtjs.org/docs/features/configuration/#pre-processors
yarn add -D sass sass-loader@10 fibers
```

But now I'm getting this error in the browser that only occurs when I use `createVuetify()` in the `plugins/vuetify.ts` file:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'DEBUG')
    at globals.ts:3
```

I think this is because it isn't transpiling `vuetify` and replacing the `process.env.DEBUG` with the actual values.. but I'm not entirely sure how that is all working.

**Reply by NotExpectedYet** [reply]
The sass issue I also had and was resolved by installing sass development dependency. 

``
Uncaught TypeError: Cannot read properties of undefined (reading 'DEBUG')
    at globals.ts:3
``

Getting the same issue myself. Fresh nuxt3 install if it helps.

**Reply by Magnie** [reply]
I should also mention, a workaround for this is to modify `node_modules/vuetify/lib/util/globals.mjs` so `IS_DEBUG` and `IS_PROD` are set to `false`:
```javascript
export const IS_NODE = typeof process !== 'undefined';
export const IN_BROWSER = typeof window !== 'undefined';
export const IS_DEBUG = false; // IS_NODE && process.?env.DEBUG === 'true';
export const IS_PROD = false; // IS_NODE && process.?env.NODE_ENV === 'production';
export const SUPPORTS_INTERSECTION = IN_BROWSER && 'IntersectionObserver' in window;
export const SUPPORTS_TOUCH = IN_BROWSER && ('ontouchstart' in window || window.navigator.maxTouchPoints > 0);
//# sourceMappingURL=globals.mjs.map
```

**Reply by Dreaming-Codes** [reply]
With this method I was unable to get the themes to work

**Reply by davidzech27** [reply]
Removing nuxt3 and installing it again worked for me. Something to do with it nuxt3 appearing in package.json under dependencies instead of devDependencies.

**Reply by jojomatik** [reply]
~~These steps work for me. The only thing not working is autocompletion for `this.$vuetify`. `TS2339: Property '$vuetify' does not exist on type [name]`. Has anyone got that working? I suppose I need to change something in `tsconfig.json`? (I'm using class style components).~~

~~The code executes fine, it's just that my IDE doesn't get it.~~

Nevermind, I just had a look at the new API and accessing the $vuetify instance isn't necessary anymore. Themes can be set using the `<v-app theme="dark">` syntax and breakpoints are accessible via the display api.

**Reply by MatthewAry** [reply]
Any suggestions on how to manipulate the Vuetify SASS variables when working with Nuxt3?

**Reply by Qendolin** [reply]
From my testing it seems that the sass for the components isn't used, instead the static, precompiled .css files are loaded. (e.g.: `node_modules\vuetify\lib\components\VField\VField.css`). I tested this by modifying the sass & css files directly but only changes in the css file had an effect.

**Reply by arinzejustin** [reply]
@jojomatik Please am having issues with setting dark mode theme with button toggle 
Can you please help me out
Thanks

**Reply by arinzejustin** [reply]
@jojomatik I have just figured it out 
Thanks

**Comment by lbiernot** [comment]
This is still not resolved and is pretty annoying.
It should be straightforward - bootstrap Nuxt3 app, add Vuetify and define plugin.
I totally understand that Nuxt3 is beta and Vuetify 3 is alfa so they aren't stable but they should be at least usable to a limited extent.

**Comment by rodrigogs** [comment]
So... not yet working?

**Comment by rodrigogs** [comment]
Well, I kinda got it working here: https://github.com/rodrigogs/nuxt3-vuetify-pug-template

The thing is... it's far from being done hahahah

Most of the components are not converted to Vue 3, so... there's no point in trying to use it for now. Anyway, this repo is a good one for tracking the progress while Vuetify team develops it, you just have to update vuetify's version from time to time.

**Comment by amery** [comment]
I'm new on this and all that pug-ness left me really confused. I'm trying something **way** simpler, just a dumb counter without anything v-* yet but it stops working as soon as I add the `transpile:`.

https://github.com/amery/nuxt3-vuetify/blob/master/src/app.vue

and without the transpile vite explodes :-/
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for /home/amery/projects/go-nuxt/src/github.com/amery/nuxt3-vuetify/node_modules/vuetify/lib/components/VCode/VCode.css
    at new NodeError (node:internal/errors:371:5)
    at Object.file: (node:internal/modules/esm/get_format:72:15)
    at defaultGetFormat (node:internal/modules/esm/get_format:85:38)
    at defaultLoad (node:internal/modules/esm/load:13:42)
    at ESMLoader.load (node:internal/modules/esm/loader:303:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:230:58)
    at new ModuleJob (node:internal/modules/esm/module_job:63:26)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:244:11)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:78:21)
    at async Promise.all (index 0) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
```

any idea what's going on?

**Comment by vincenthalman** [comment]
Vuetify is still in development to get to Vue 3. But is there any other VUE 3 component library that we could use within nuxt3? I know nuxt3 is beta and we could use nuxt2 instead but I'm starting a new project and like to make it in typescript and the composition api.

**Reply by rodrigogs** [reply]
> Vuetify is still in development to get to Vue 3. But is there any other VUE 3 component library that we could use within nuxt3? I know nuxt3 is beta and we could use nuxt2 instead but I'm starting a new project and like to make it in typescript and the composition api.

You could try nuxt bridge.

**Reply by cmmata** [reply]
In my company, we are using nuxt3 with [Element Plus](https://element-plus.org/en-US/). It is working quite well if we know that nuxt3 is still beta.

**Reply by vincenthalman** [reply]
Oh that would be nice! I wonder how you've been able to implement Element plus within Nuxt3. I don't get it to work either...

**Reply by cmmata** [reply]
@vincenthalman I have this opened bug when I try to use the carousel https://github.com/element-plus/element-plus/issues/5030

But if you want to take a look on how is it implemented, here's a working example. Most of the Element components are working fine https://github.com/cmmata/nuxt3-elementPlus

**Comment by KaelWD** [comment]
This is how to make it work with the vuetify vite plugin:

`yarn add -D vuetify@next vite-plugin-vuetify`

nuxt.config.ts:
```ts
import { defineNuxtConfig } from 'nuxt3'
import vuetify from 'vite-plugin-vuetify'

export default defineNuxtConfig({
  css: ['vuetify/styles'],
  vite: {
    ssr: {
      noExternal: ['vuetify'],
    },
  },
  modules: [
    async (options, nuxt) => {
      nuxt.hooks.hook('vite:extendConfig', config => config.plugins.push(
        vuetify()
      ))
    }
  ]
})
```

plugins/vuetify.ts:
```ts
import { defineNuxtPlugin } from '#app'
import { createVuetify } from 'vuetify'

export default defineNuxtPlugin(app => {
  const vuetify = createVuetify({ ssr: true })

  app.vueApp.use(vuetify)
})
```

**Reply by etino** [reply]
Hello @KaelWD 
trying to implement your solution I get a typescript error into the `configResolved` function that reorders plugins.

`config` argument is of type `ResolvedConfig` that has `plugins` property flagged as `Readonly` so the `splice` method is not available.

Any workaround?

[Reference from Vite Repository](https://github.com/vitejs/vite/blob/a34561490b4b866d8d4f98c697435dcb68a5c3ed/packages/vite/src/node/config.ts#L254)
```ts
export type ResolvedConfig = Readonly<
  Omit<
    UserConfig,
    'plugins' | 'alias' | 'dedupe' | 'assetsInclude' | 'optimizeDeps' | 'worker'
  > & {
    configFile: string | undefined
    configFileDependencies: string[]
    inlineConfig: InlineConfig
    root: string
    base: string
    publicDir: string
    command: 'build' | 'serve'
    mode: string
    isProduction: boolean
    env: Record<string, any>
    resolve: ResolveOptions & {
      alias: Alias[]
    }
    plugins: readonly Plugin[]
    server: ResolvedServerOptions
    build: ResolvedBuildOptions
    preview: ResolvedPreviewOptions
    assetsInclude: (file: string) => boolean
    logger: Logger
    createResolver: (options?: Partial<InternalResolveOptions>) => ResolveFn
    optimizeDeps: Omit<DepOptimizationOptions, 'keepNames'>
    /** @internal */
    packageCache: PackageCache
    worker: ResolveWorkerOptions
  }
>
```

**Reply by wobsoriano** [reply]
@etino  Do you also get this error with [this solution](https://github.com/nuxt/framework/discussions/1183#discussioncomment-1539889)?

**Reply by etino** [reply]
> @etino Do you also get this error with [this solution](https://github.com/nuxt/framework/discussions/1183#discussioncomment-1539889)?

No. The solution you quote works. 
I tested it after posting my message noticing that it was edited recently. I was in trouble seeing the original posting date.
I think [it is the right answer to this discussion](https://github.com/nuxt/framework/discussions/1183#discussioncomment-1539889).

**Reply by rodrigogs** [reply]
Vuetify is not ready for vue 3, most of the components are not going to work anyway.

**Reply by KaelWD** [reply]
The types are wrong then, because it runs fine. It's also only necessary because of #2798

**Reply by brownsugar** [reply]
Hi @KaelWD 
There is an option called [enforce](https://vitejs.dev/guide/api-plugin.html#plugin-ordering) for vite plugins, I think the import plugin in `vite-plugin-vuetify` can use `enforce: 'post'` directly instead of using `configResolved` in nuxt config.

**Reply by KaelWD** [reply]
Nope, it needs to run after Vue but before Vite build plugins.

**Reply by brownsugar** [reply]
Hi and thanks, but I got an error even I set the `configResolved` option while using `vite-plugin-vuetify@1.0.0-alpha.11`:
<img width="750" alt="image" src="https://user-images.githubusercontent.com/2935980/171084577-5155f0d9-f915-4cad-a75f-83f839a802d0.png">

I have to roll back to `@vuetify/vite-plugin@1.0.0-alpha.10` to run this normally.

Environment:
 - Node.js 16
 - vite-plugin-vuetify@^1.0.0-alpha.11
 - nuxt@^3.0.0-rc.3
 - vite@^2.9.9

**Reply by markgibbons25** [reply]
Referencing https://github.com/vuetifyjs/vuetify/issues/14984 as it looks like this just needs to be added to the documentation as the way to configure.

**Reply by kotx** [reply]
Should we add `ssr: true` to `createVuetify({})`?

**Reply by rrolla** [reply]
Thanks @KaelWD looks like treeshaking now works :)

**Comment by JPustkuchen** [comment]
Good morning, is the Vuetify or the Nuxt3 Team planning to create an official integration / documentation for the Nuxt3+Vuetify integration? I think both would be a fabulous team, but currently it seems quite hard to get them combined?

**Comment by Zebnastien** [comment]
I could get Vuetify Beta to work witch Nuxt 3 thanks to RazorCX and his tutorial:
https://www.youtube.com/watch?v=DE3HABBzUWY

Now I am looking forward to use Vuetify integrated mdi icons.

**Reply by cbrhex** [reply]
```
yarn add mdi
```

```
import {defineNuxtConfig} from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
    css: [
        ....
        'mdi/css/materialdesignicons.min.css',
        ...
    ],
})
```


```
      <v-btn
          icon
          color="primary"
      >
        <v-icon>mdi-heart</v-icon>
      </v-btn>
```

**Reply by Zebnastien** [reply]
It works, thanks !

**Reply by Caceresenzo** [reply]
The `mdi` package has been [deprecated](https://www.npmjs.com/package/mdi).
You should now use: [`@mdi/font`](https://www.npmjs.com/package/@mdi/font).

Here are what need to be changed:

```bash
npm install @mdi/font
```

```js
export default defineNuxtConfig({
    css: [
        ....
        '@mdi/font/css/materialdesignicons.min.css',
        ...
    ],
})
```

**Reply by Ilmion** [reply]
When adding materialdesignicons.min.css to nuxt config css, the icons work, but I get this warning multiple time in the console :
`[Vue Router warn]: No match found for location with path "/materialdesignicons.min.css.map"`

**Comment by CodyBontecou** [comment]
If you're interested, I did a quick write-up on getting this to work [here](https://codybontecou.com/how-to-use-vuetify-with-nuxt-3.html).

The key points:
1. `yarn add vuetify@next sass`
2. Create the following vuetify plugin:
```js
// plugins/vuetify.js
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export default defineNuxtPlugin(nuxtApp => {
  const vuetify = createVuetify({
    components,
    directives,
  })

  nuxtApp.vueApp.use(vuetify)
})
```
3. Configure your nuxt config:
```js
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  css: ['vuetify/lib/styles/main.sass'],
  build: {
    transpile: ['vuetify'],
  },
  vite: {
    define: {
      'process.env.DEBUG': false,
    },
  },
})
```

**Comment by JPustkuchen** [comment]
Thanks @CodyBontecou, perhaps contact the Vuetify team to put something into the Vuetify 3 official documentation? Or provide an official nuxt3 integration Vuetify plugin to further simplify things?

**Reply by CodyBontecou** [reply]
I'll see if I can get a PR integrated into the Vuetify docs. As for a dedicated Nuxt 3 plugin, that is a good idea. I'll look into what that takes.

@JPustkuchen Do you know of a document that may help with me building out a publishable plugin?

**Comment by schoenix** [comment]
Maybe a problem with server-side and client-side rendering.

**Comment by Kingside88** [comment]
I also managed this to work. Thanks to RazorCX Video. 
But it takes so long to run npm run dev. Over 1 minute.

**Comment by amery** [comment]
I wrote a tiny test for nuxt@rc + vuetify@next but it still doesn't work. I have no idea what I'm doing wrong. help?

https://stackblitz.com/github/amery/nuxt3-vuetify?file=src%2Flayouts%2Fvuetify.vue

**Reply by CodyBontecou** [reply]
Change your `nuxt.config.ts` to look something like:

```js
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt';

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  css: ['vuetify/lib/styles/main.sass'],
  build: {
    transpile: ['vuetify'],
  },
  vite: {
    define: {
      'process.env.DEBUG': false,
    },
  },
});

```

**Reply by amery** [reply]
thanks, I can confirm defining `process.env.DEBUG` as described by @CodyBontecou  fixes the problem

**Comment by tadvas** [comment]
Has anyone successfully integrated SCSS Custom Variables?
Adding 
```
vuetify: {
  customVariables: ['~/assets/variables.scss'],
  treeShake: true
}
```
did not do the job :(

**Reply by davidramosweb** [reply]
Hi, you can do it as follows:

1. Create a main.scss file and load the vuetify module overwriting the variables like this:
`/assets/main.scss`
```
@use 'vuetify/styles/main' with (
    $border-radius-root: 10px,
    );
```

2. Use this file in your `nuxt.config.ts`

```
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt';

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
  css: ['~/assets/main.scss'],
```

**Reply by tadvas** [reply]
Hey David, thanks for reply! The solution looks great and it works. However how can we use `@use` and `@import` together in .scss file? Import is needed for fonts, for example.

**Reply by skaaptjop** [reply]
I think this might need to be updated to `vuetify/lib/styles/main` instead of `vuetify/styles/main`

**Comment by swissbuechi** [comment]
Does anyone know how to access `this.$vuetify` in nuxt3?
Thanks!

**Reply by rodrigogs** [reply]
Since `Vue 3` `script setup` tags have no access to `this`, I don't think you will be able to call `this.$vuetify` inside a `Nuxt 3` component.

**Reply by KaelWD** [reply]
They're separate composables now https://next.vuetifyjs.com/en/features/theme/#changing-theme

**Reply by swissbuechi** [reply]
Thank you very much! @KaelWD 
My goal is to change the theme with a button.
I'm now able to retrieve the current Theme but I can not change it.

```typescript
<script lang="ts" setup>
import { useTheme } from 'vuetify'

const theme = useTheme()

function toggleTheme(){
  theme.name.value = theme.current.value.dark ? 'light' : 'dark'
}

</script>
```

Error from browser console:

`Write operation failed: computed value is readonly`

I would appreciate any help.
Thanks a lot.

**Reply by KaelWD** [reply]
Ah damn, seems like we overlooked setting the theme from a not-root-level component

**Reply by swissbuechi** [reply]
Thanks for your answer.
So it should work if I change it for example in the default.vue layout?
Right now I tried to change it from my custom AppBar.vue component.

**Reply by Moutah** [reply]
Had the exact same configuration: wanted to toggle the theme from my `AppBar.vue` but had the same error message.

I could go around it by emitting an output from `AppBar.vue` and handling it in the main `app.vue`.

While this works, it doesn't feel ideal. @KaelWD do you think it'll be possible to allow any component to set the theme ?

---

(simplified code)

`AppBar.vue`
```vue
<script lang="ts" setup>
import { useTheme } from 'vuetify';

const theme = useTheme();

const emit = defineEmits<{ (e: 'theme-changed', value: string): void }>();

const themeIcon = computed(() =>
  theme.current.value.dark ? 'white-balance-sunny' : 'moon-waning-crescent',
);

function toggleTheme() {
  emit('theme-changed', theme.current.value.dark ? 'madmsLightTheme' : 'madmsDarkTheme');
}
</script>

<template>
  <v-app-bar>

    <v-btn variant="text" @click="toggleTheme">
      <v-icon>{{ `mdi-${themeIcon}` }}</v-icon>
    </v-btn>

  </v-app-bar>
</template>
```

`app.vue`
```vue
<script setup lang="ts">
import { useTheme } from 'vuetify';

const theme = useTheme();
</script>

<template>
  <v-app>
    <AppBar @theme-changed="theme.name.value = $event" />
    
    <v-main>
      😎
    </v-main>
  </v-app>
</template>
```

**Reply by KaelWD** [reply]
Yes I've added `theme.global`

**Comment by shtse8** [comment]
I can't get it to work using runtime template with vuetify at the same time.

Everything is working fine except runtime template before adding:
```
hooks: {
    'vite:extendConfig': (config, { isClient, isServer }) => {
      if (isClient)
        config.resolve.alias.vue = 'vue/dist/vue.esm-bundler.js'
    },
  },
```

Now, my runtime template is working great! but my vuetify has gone.

**Comment by kstraszewski** [comment]
Hi, I've discovered that adding vuetify to nuxt3 kills FCP in production mode. Here is an example of my project that uses vuetify:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/10195995/186275769-8e7c92ae-74a8-4acf-b630-0e7b08223bfb.png">
You can check it [here](https://demo.lendi.pl/test) for your own.
FCP is blocked by huge entry.css file after build. And that creates an enormous entry.css and performance is very bad because of it 

Is there any way to tree shake CSS from entry.css file in production mode?

**Reply by stephanUllenboom** [reply]
Is the FCP issue already solved?

**Reply by CodyBontecou** [reply]
Have you guys tried this method of adding Vuetify, rather than using a plugin?

```js
import vuetify from 'vite-plugin-vuetify'

export default defineNuxtConfig({
  modules: [
    async (options, nuxt) => {
      nuxt.hooks.hook('vite:extendConfig', config => config.plugins.push(
        vuetify({
          styles: { configFile: 'src/settings.scss' })
      ))
    }
  ],
})
```

It's mentioned in their docs here - https://next.vuetifyjs.com/en/features/treeshaking/

**Reply by stephanUllenboom** [reply]
Thanks. @CodyBontecou I tried this method and works for me. 

I think in August the article was not yet available.

EDIT:
Small code change, you miss one curly bracket in the vuetify configuration:
```js
import vuetify from 'vite-plugin-vuetify'

export default defineNuxtConfig({
  modules: [
    async (options, nuxt) => {
      nuxt.hooks.hook('vite:extendConfig', (config) =>
        config.plugins.push(
          vuetify({
            styles: { configFile: 'src/settings.scss' },
          })
        )
      );
    },
  ],
});
```

**Comment by JPustkuchen** [comment]
Once we are sure we know, how it works, someone should please add a merge request for an official NUXT integration documentation page at Vuetify, I think?

https://github.com/vuetifyjs/vuetify/issues/12271

**Comment by sakrlog** [comment]
Someone already mentioned this in here but its been a while. Vuetify's theme configuration is not taking effect but the default one is. I have `ssr: true` and sass-loader installed. One thing to notice, theme setting is working between dark and light but the colors aren't.

**Reply by rrolla** [reply]
for me it works, note its `typescript`
```ts
const myCustomDarkTheme: ThemeDefinition = {
  dark: true,
  colors: {
    background: '#121212',
    surface: '#212121',
    'surface-variant': '#29b6f6',
    'on-surface-variant': '#424242',
    primary: '#29b6f6',
    'primary-darken-1': '#000000',
    secondary: '#d7ccc8',
    'secondary-darken-1': '#0086c3',
    error: '#CF6679',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00',
  },
}

export default defineNuxtPlugin((nuxtApp) => {
  const vuetify = createVuetify({
    ssr: true,
    defaults: {
      global: {
        ripple: true,
      },
      VSheet: {
        elevation: 4,
      },
    },
    locale: {
      locale: 'en',
      fallback: 'en',
    },
    theme: {
      defaultTheme: 'myCustomDarkTheme',
      themes: {
        myCustomDarkTheme,
      },
    },
  })

  nuxtApp.vueApp.use(vuetify)
})
```
